platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_performance.adoc

```diff
@@ -42,3 +42,47 @@ NOTE: Do not rely on the default encoding of the underlying JVM or operating sys
 === Tags
 
 `<<tags_reference.adoc#tag:properties,#properties>>`, `<<tags_reference.adoc#tag:caching,#caching>>`
+
+[[avoid_computations_in_configuration_phase]]
+== Avoid Expensive Computations in Configuration Phase
+
+Avoid expensive computations in the <<build_lifecycle.adoc#sec:build_phases,configuration phase>>, instead, move them to task actions.
+
+=== Explanation
+
+In order for Gradle to execute tasks it first needs to build the project task graph. This work is done in a configuration phase of the build. Even if you run a single task, the entire task graph needs to be created.
+
+Adding expensive computations such as doing I/O, loading large files, or CPU-heavy operations in configuration phase can force these to be run when they could be avoided. It is better to move this work to task actions that would run these only when required.
```

Comment:
```suggestion
Peforming expensive or slow operations such as file or network I/O, or CPU-heavy calculations in the configuration phase forces these to run even when they might be unnecessary to complete the requested work of the invoked tasks.
It is better to move these operations to task actions so that they run only when required.
```